cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -976,6 +983,32 @@ public void getServiceInstances(HttpRequest request, HttpResponder responder,
   }
 
   /**
+   * Return the
+   */
+  @GET
+  @Path("/apps/{app-id}/services/{service-id}/active")
+  public void getServiceStatus(HttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId,
+                               @PathParam("app-id") String appId,
+                               @PathParam("service-id") String serviceId) throws NotFoundException {
+    ProgramId programId = Ids.namespace(namespaceId).app(appId).service(serviceId);
+    ProgramStatus status = lifecycleService.getProgramStatus(programId);
```

Comment:
should this check whether the program exists? It looks like lifecycleService does not, it only checks existence of the app. Perhaps add it there.